<a href="https://colab.research.google.com/github/Liliana2233/Tareas/blob/main/tarea_02_pandas_ingestion.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Tarea 02 — Carga de Datos con Pandas

**Instrucciones:** Completa cada celda de código con tu respuesta.  
Usarás el dataset de Iris disponible en la URL indicada.

In [ ]:
import pandas as pd

---
## Parte 1 — Cargar y explorar datos

**P1.** Carga el dataset de Iris desde la siguiente URL y guárdalo en una variable llamada `df`.  
URL: `https://raw.githubusercontent.com/mwaskom/seaborn-data/master/iris.csv`

In [4]:
url = "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/iris.csv"

# Tu respuesta aquí
import pandas as pd

df = pd.read_csv(url)
df

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa
...,...,...,...,...,...
145,6.7,3.0,5.2,2.3,virginica
146,6.3,2.5,5.0,1.9,virginica
147,6.5,3.0,5.2,2.0,virginica
148,6.2,3.4,5.4,2.3,virginica


**P2.** Muestra las primeras 5 filas del DataFrame.

In [5]:
# Tu respuesta aquí
df.head()

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


**P3.** Usa `.info()` para ver los tipos de columnas y si hay valores nulos.  
Luego usa `.describe()` para ver el resumen estadístico. ¿Cuántas filas tiene el dataset?

In [7]:
# Tu respuesta aquí

df.info()
df.describe()
df.shape[0]

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


150

**Respuesta:** El dataset tiene 150 filas.

---
## Parte 2 — Selección y filtrado

**P4.** Selecciona solo las columnas `sepal_length` y `species` e imprime las primeras 5 filas.

In [8]:
# Tu respuesta aquí
df[["sepal_length", "species"]].head()

,sepal_length,species
0,5.1,setosa
1,4.9,setosa
2,4.7,setosa
3,4.6,setosa
4,5.0,setosa


**P5.** Filtra el DataFrame para quedarte solo con las flores de la especie `virginica`.  
¿Cuántas filas tiene ese subconjunto?

In [17]:
# Tu respuesta aquí

df_virginica = df[df["species"] == "virginica"]

df_virginica.head()
df_virginica.shape[0]



50

**P6.** Filtra las filas donde `petal_length` sea mayor a 5.0. Imprime cuántas filas cumplen esa condición.

In [11]:
# Tu respuesta aquí
df_petal_length = df[df["petal_length"] > 5.0]

df_petal_length.shape[0]

42

---
## Parte 3 — Guardar datos

**P7.** Guarda el subconjunto de `virginica` (de P5) en un archivo CSV llamado `virginica.csv` en la carpeta `output/`.  
No incluyas el índice en el archivo guardado.

In [13]:
import os
os.makedirs("output", exist_ok=True)

# Tu respuesta aquí
df_virginica.to_csv("output/virginica.csv", index=False)

**P8.** Carga de nuevo el archivo `output/virginica.csv` y verifica que tiene el número correcto de filas usando `.shape`.

In [14]:
# Tu respuesta aquí
df_virginica_nuevo = pd.read_csv("output/virginica.csv")

df_virginica_nuevo.shape

(50, 5)